sequencer/src/bytes.rs

```diff
@@ -0,0 +1,193 @@
+//! Convenient serialization for binary blobs.
+
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use base64::{engine::general_purpose::STANDARD as BASE64, Engine};
+use derive_more::{From, Into};
+use serde::{
+    de::{Deserialize, Deserializer, Error},
+    ser::{Serialize, Serializer},
+};
+use std::{
+    ops::{Deref, DerefMut},
+    slice::SliceIndex,
+};
+
+// Re-export `bytes![]` macro, which gets exported at the crate root by default.
+pub use crate::bytes;
+
+/// An unstructured byte array with smart serialization.
+///
+/// [`Bytes`] mostly acts as a simple byte array, `Vec<u8>`. It can easily be converted to and from
+/// a `Vec<u8>`, and it implements many of the same traits as [`Vec`]. In fact internally it merely
+/// wraps a `Vec<u8>`.
+///
+/// The only difference is in how it serializes. `Vec<u8>` serializes very efficiently using
+/// `bincode`, but using `serde_json`, it serializes as a JSON array of integers, which is
+/// unconventional and inefficient. It is better, in JSON, to serialize binary data as a
+/// base64-encoded string. [`Bytes`] uses the [`is_human_readable`](Serializer::is_human_readable)
+/// property of a [`Serializer`] to detect whether we are serializing for a compact binary format
+/// (like `bincode`) or a human-readable format (like JSON). In the former cases, it serializes
+/// directly as an array of bytes. In the latter case, it serializes as a string using base 64.
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    From,
+    Into,
+    CanonicalSerialize,
+    CanonicalDeserialize,
+)]
+pub struct Bytes(Vec<u8>);
```

Comment:
Please don't block merge for this comment, but I wonder if going forward we can find a less generic name to avoid confusion w/ [`bytes::Bytes`](https://docs.rs/bytes/latest/bytes/struct.Bytes.html).